sts_auth/cli.py

```diff
@@ -73,21 +71,29 @@ def authenticate(username, password, idpentryurl, domain,
 
     sts_auth.parse_config_file()
 
-    assertion = sts_auth.get_saml_response()
+    saml_response = sts_auth.get_saml_response()
+    adfs_response = sts_auth.fetch_aws_account_names(saml_response)
+    account_map = utils.parse_aws_account_names_from_response(adfs_response)
     # Parse the returned assertion and extract the authorized roles
-    awsroles = utils.parse_roles_from_assertion(assertion)
-    account_roles = utils.format_roles_for_display(awsroles)
+    awsroles = utils.parse_roles_from_assertion(saml_response.assertion)
+    account_roles = utils.format_roles_for_display(awsroles, account_map)
+    account_roles_len = len(account_roles)
+    account_roles_vals_len = len(list(account_roles.values())[0])
```

Comment:
You're getting the values, making a list, then getting the length of the first value in the list?

Reply:
In the case that a user has access to only one account and only one role in that account, we want to bypass prompting them to select it and just automatically fetch credentials for it. So we get the `account_roles.values()` which are, in this case, a list of lists containing the roles in accounts. (`account_roles.keys()` would be a list of account ids) and make sure that for the first item in the list, there is only one item in that list.